packages/@aws-cdk/core/lib/storage.ts

```diff
@@ -0,0 +1,135 @@
+import { Token } from "./token";
+
+/**
+ * Represents the amount of digital storage.
+ *
+ * The amount can be specified either as a literal value (e.g: `10`) which
+ * cannot be negative, or as an unresolved number token.
+ *
+ * When the amount is passed as a token, unit conversion is not possible.
+ */
+export class Storage {
+  /**
+   * Create a Storage representing an amount kibibytes.
+   */
+  public static kibibytes(amount: number): Storage {
```

Comment:
Can we guarantee that they're actually kibibytes and not kilobytes in all services?

If not, does it even make sense to be so explicit? Shouldn't we rather use `kilobytes` and accepting the ambiguity?

Reply:
I spot checked and quite a few places in the documentation, clearly uses the unit as M*i*B or G*i*B. However, there are other places that use the terms "MB" and "GB" but my gut feeling says that they still using these terms while they actually mean MiB and GiB.

For instance, take the [`AllocatedStorage` property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-allocatedstorage), the description explicitly says "gigabytes". However, in the [first example here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#aws-properties-rds-database-instance--examples), the description uses "GiB".

> Shouldn't we rather use kilobytes and accepting the ambiguity?

Isn't that more confusing? If we do come across an instance where kilobytes or megabytes are used, wouldn't it be clearer to encode these as further units of `Storage`?

Reply:
In any case I think a lot of people actually live through the day without knowing the difference, or struggling to remember which is which. Feels like the docstrings should include some information with respects to that (aka - 1KiB is 1024B, 1MiB is 1024KiB, ...). If/when we start throwing kB, we'd say 1kB is 1000B, etc...

Reply:
@RomainMuller  - I've updated. Does this satisfy?

Reply:
@nija yes it does!